components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -79,10 +105,27 @@
                 if (handlerElem.getAttribute(CLASS_Q) != null) {
                     String className = handlerElem.getAttributeValue(CLASS_Q);
                     if (!className.isEmpty()) {
-                        InternalAPI internalApi = createAPI(className);
+                        InternalAPI internalApi = createApi(className);
                         if (internalApi != null) {
-                            internalApis.add(internalApi);
                             internalApi.setName(name);
+                            if (handlerElem.getAttribute(PROTOCOL_Q) != null) {
+                                String protocols = handlerElem.getAttributeValue(PROTOCOL_Q);
+                                if (!protocols.isEmpty()) {
+                                    String[] protocolList = protocols.split(" ");
+                                    for (String protocol : protocolList) {
+                                        switch (protocol) {
+                                        case "http":
+                                            httpInternalAPIList.add(internalApi);
+                                            break;
+                                        case "https":
+                                            httpsInternalAPIList.add(internalApi);
+                                            break;
+                                        default:
+                                            handleException("Unsupported Protocol found for Internal API");
+                                        }
+                                    }
```

Comment:
We might need to have a WARN log saying not protocols are enabled for the API?

Reply:
added